sched/scheduler/stateful_scheduler.go

```diff
@@ -578,26 +576,33 @@ addLoop:
 		select {
 		case newJobMsg := <-s.addJobCh:
 			receivedJob = true
-
+			lf := log.Fields{
+				"jobID":     newJobMsg.job.Id,
+				"requestor": newJobMsg.job.Def.Requestor,
+				"jobType":   newJobMsg.job.Def.JobType,
+				"tag":       newJobMsg.job.Def.Tag,
+				"basis":     newJobMsg.job.Def.Basis,
+				"priority":  newJobMsg.job.Def.Priority,
+				"numTasks":  len(newJobMsg.job.Def.Tasks),
+			}
 			if _, ok := s.requestorMap[newJobMsg.job.Def.Requestor]; ok {
-				// If we have an existing job with this requestor/tag combination, make sure we use its priority level.
-				// Not an error since we can consider priority to be a suggestion which we'll handle contextually.
 				for _, js := range s.requestorMap[newJobMsg.job.Def.Requestor] {
+					// Kill any prior jobs that share the same requestor and non-empty basis string,
+					// but tag should be different so we don't kill the original jobs when doing retries.
```

Comment:
Could you expand on this point? Do retries get assigned a non-unique "retry" tag?